src/rviz/robot/robot_link.cpp

```diff
@@ -595,22 +616,22 @@ void RobotLink::createEntityForGeometryElement(const urdf::LinkConstSharedPtr& l
 
     scale = Ogre::Vector3(mesh.scale.x, mesh.scale.y, mesh.scale.z);
 
-    std::string model_name = mesh.filename;
+    const std::string& model_name = mesh.filename;
 
     try
     {
-      loadMeshFromResource(model_name);
-      entity = scene_manager_->createEntity(ss.str(), model_name);
+      if (loadMeshFromResource(model_name).isNull())
+        addError("Could not load mesh resource '%s'", model_name.c_str());
```

Comment:
```suggestion
        addError(QString("Could not load mesh resource '%1'").arg(model_name.c_str()));
```
or so, just a suggestion to keep things simple